python/paddle/fluid/dygraph/dygraph_to_static/ifelse_transformer.py

```diff
@@ -183,37 +222,119 @@ def visit_Name(self, node):
             if wrapper_node.node_var_type & {
                     NodeVarType.TENSOR, NodeVarType.PADDLE_RETURN_TYPES
             }:
-                self.is_control_flow = True
+                self.is_control_flow_num += 1
         return node
 
-    def _is_candidate_node(self, node):
-        return isinstance(node, (gast.Compare, gast.BoolOp))
-
-    def _compare_with_none(self, node):
-        if isinstance(node, gast.Compare):
-            for child in [node.left, node.comparators]:
-                # node.comparators is a list.
-                if isinstance(child, list):
-                    child = child[0]
-                if (isinstance(child, gast.Constant) and
-                        child.value is None) or (
-                            isinstance(child, gast.Name) and
-                            child.id == 'None'):
-                    return True
-        return False
+    def get_compare_nodes_with_tensor(self):
+        return self._compare_node_tenor_set
 
 
-def is_control_flow_if(node, static_analysis_visitor=None):
-    """
-    Determine whether the node is a plain python `if statement` or
-    control flow in Paddle.
-    """
-    assert isinstance(
-        node, gast.AST
-    ), "Type of input node should be gast.AST, but received %s." % type(node)
-    if static_analysis_visitor is None:
-        static_analysis_visitor = StaticAnalysisVisitor(node)
-    return IsControlFlowIfVisitor(static_analysis_visitor).transform(node)
+class IfNodeTestTransformer(gast.NodeTransformer):
```

Comment:
Can you make the IfNodeTestTransformer not only change the condition of `if` clause, but all BoolOp? Because BoolOp may be also used at `while` and common operations.

Reply:
You are right. Now I refine the class  `IsControlFlowVisitor` to make it judge whether a node is dependent on `Tensor` separately.